### PR TITLE
Build out all deals page

### PR DESCRIPTION
**Pull request by vipergtsrz**
Built out the all deals page
Currency formatter was added with a optional decimal places param.

Unit tests were created for the all deals page

### CODE REVIEW

File: src/resources/elements/ConnectButton/ConnectButton.ts [hunk 1]
```diff
@@ -58,7 +60,9 @@ export class ConnectButton {
       this.txPhase = Phase.None;
       this.txReceipt = null;
     }));
-
+    this.subscriptions.push(this.eventAggregator.subscribe("account.connect", async () => {
```
Comment (dkent600) [inline src/resources/elements/ConnectButton/ConnectButton.ts:63]: why is this needed?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/deals/list/deals.ts [hunk 2]
```diff
@@ -1,78 +1,161 @@
...
+    }
+  }
+
+  /**
+   * Returns a filtered set of deals based on criteria being passed in from the deal service
+   * @param cardIndex The current Tab
+   * @param showMine Whether or not to show your current deals
+   * @returns
+   */
+  private getDealsForCardIndex(cardIndex: number, showMine: boolean) {
```
Comment (dkent600) [inline src/deals/list/deals.ts:118]: This should be implemented in the DealsService, not here.

File: src/resources/elements/ConnectButton/ConnectButton.ts (hunk 1 above)
Comment (vipergtsrz) [reply src/resources/elements/ConnectButton/ConnectButton.ts:63]: When the user's wallet isn't connected, the "onConnect" method in the deals.ts code calls 

`this.eventAggregator.publish("account.connect");`

Which triggers the exact same code as the 'onConnect' method inside the ConnectButton.ts. The code you're referring to is needed to allow the deals.ts code to connect their wallet without having to duplicate code.

File: src/deals/list/deals.ts [hunk 3]
```diff
@@ -1,78 +1,161 @@
+import "./deals.scss";
+
+import { autoinject, computedFrom, singleton } from "aurelia-framework";
+
+import { DateService } from "services/DateService";
+import { DealService } from "services/DealService";
```
Comment (dkent600) [inline src/deals/list/deals.ts:6]: This will create a module reference loop.

File: src/deals/list/deals.ts (hunk 2 above)
Comment (dkent600) [reply src/deals/list/deals.ts:118]: Also should have a return value

Comment (dkent600) [inline src/deals/list/deals.ts:118]: All methods should declare their return type.

**Review by dkent600 — CHANGES_REQUESTED** [review]
Sort by DAO and Title

**Comment by dkent600** [comment]
When we change accounts while "Show My Deals Only", we should recompute the array

**Review by dkent600 — APPROVED** [review]
